Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -150,4 +151,105 @@ public static int CountSubstring(this string str, string substring, StringCompar
 
         return count;
     }
+
+    /// <summary>
+    /// Determines if the <paramref name="value"/> is longer than 8 characters or contains an <see cref="Environment.NewLine"/>.
+    /// </summary>
+    public static bool IsLongOrMultiline(this string value)
+    {
+        const int humanReadableLength = 8;
+        return value.Length > humanReadableLength || value.Contains(Environment.NewLine, StringComparison.Ordinal);
+    }
+
+    /// <summary>
+    /// Get the mismatch segment between <paramref name="expected"/> and <paramref name="subject"/> for long strings,
+    /// when they differ at index <paramref name="firstIndexOfMismatch"/>.
+    /// </summary>
+    public static string GetMismatchSegmentForLongStrings(this string subject, string expected, int firstIndexOfMismatch)
+    {
+        int trimStart = CalculateSegmentStart(subject, firstIndexOfMismatch);
+
+        int whiteSpaceCount = (firstIndexOfMismatch - trimStart) + 3;
+        const string prefix = "  ";
+
+        if (trimStart > 0)
+        {
+            whiteSpaceCount++;
+        }
+
+        var visibleText = subject.Substring(trimStart, firstIndexOfMismatch - trimStart);
+        whiteSpaceCount += visibleText.Count(c => c is '\r' or '\n');
+
+        var sb = new StringBuilder();
+
+        sb.Append(' ', whiteSpaceCount).AppendLine("\u2193 (actual)")
+            .Append(prefix).Append('\"').AppendVisibleText(subject, trimStart).Append('\"').AppendLine()
+            .Append(prefix).Append('\"').AppendVisibleText(expected, trimStart).Append('\"').AppendLine()
+            .Append(' ', whiteSpaceCount).Append("\u2191 (expected)");
+
+        return sb.ToString();
+    }
+
+    private static StringBuilder AppendVisibleText(this StringBuilder sb, string subject, int trimStart)
+    {
+        var subjectLength = CalculateSegmentLength(subject[trimStart..]);
+
+        if (trimStart > 0)
+        {
+            sb.Append('\u2026');
+        }
+
+        sb.Append(subject
+            .Substring(trimStart, subjectLength)
+            .Replace("\r", "\\r", StringComparison.OrdinalIgnoreCase)
+            .Replace("\n", "\\n", StringComparison.OrdinalIgnoreCase));
+
+        if (subject.Length > (trimStart + subjectLength))
+        {
+            sb.Append('\u2026');
+        }
+
+        return sb;
+    }
+
+    /// <summary>
+    /// Calculates the start index of the visible segment from <paramref name="value"/> when highlighting the difference at <paramref name="index"/>.<br />
+    /// Either keep the last 10 characters before <paramref name="index"/> or a word begin (separated by whitespace) between 15 and 5 characters before <paramref name="index"/>.
+    /// </summary>
+    private static int CalculateSegmentStart(string value, int index)
+    {
+        if (index <= 10)
+        {
+            return 0;
+        }
+
+        var wordSearchBegin = Math.Max(index - 16, 0);
```

Comment:
🔧 You refer to `15` in the docs, so use a `const` to show how you got to `16.
🔧 Rename `wordSearchBegin` to something like `indexToStartSearchingForWordBoundary`

Reply:
Used `const` values for `defaultLength`, `minLength`, `maxLength` and `lengthOfWhitespace` in [9becb00](https://github.com/fluentassertions/fluentassertions/pull/2307/commits/9becb00b59cf8e3fcb7ed2bfdec5b05582b2b44b).